File: LeetCode Solutions/Problems/2383. Minimum Hours of Training to Win a Competition.java
class Solution {
    public int minNumberOfHours(int initialEnergy, int initialExperience, int[] energy, int[] experience) {
        int answer = 0;
        for (int i=0;i<energy.length;i++){
            if (initialEnergy <= energy[i]){
                answer += energy[i]-initialEnergy+1;
                initialEnergy = 1;
            }
            else{
                initialEnergy = initialEnergy-energy[i];
            }
            if (initialExperience <= experience[i]){
                answer += experience[i]-initialExperience+1;
                initialExperience = experience[i]+1;
            }
            initialExperience = experience[i]+initialExperience;
        }
        return answer;
    }
}

File: LeetCode Solutions/Problems/2383. Minimum Hours of Training to Win a Competition.py
class Solution:
    def minNumberOfHours(self, initialEnergy: int, initialExperience: int, energy: List[int], experience: List[int]) -> int:
        currentEnergy = initialEnergy
        currentExperience = initialExperience
        answer = 0
        for i in range(len(energy)):
            if currentExperience <= experience[i]:
                answer += experience[i] - currentExperience+1
                currentExperience = experience[i]+1
            if currentEnergy <= energy[i]:
                answer += energy[i] - currentEnergy + 1
                currentEnergy = energy[i]+1
            currentEnergy -= energy[i]
            currentExperience += experience[i]
        return answer